DSCResources/cChocoPackageInstall/cChocoPackageInstall.psm1

```diff
@@ -550,4 +573,18 @@ function Get-ChocoVersion {
     Return $res
 }
 
+function Get-VersionCore {
+    [CmdletBinding()]
+    param (
+        [string]$Version
+    )
+
+    [Version] $versionCore = $null
+    if ($Version -match '(?<Major>[1-9]\d*)(\.(?<Minor>[0-9]*))?(\.(?<Patch>[0-9]*))?(\.(?<Segment>[0-9]*))?([-+].*)?') {
+        $versionCore = New-Object System.Version($Matches.Major, $Matches.Minor, $Matches.Patch, $Matches.Segment)
+    }
+
+    $versionCore
+}

```

Comment:
This doesn't compare prerelease versions so 3.0.0-beta1 and 3.0.0-beta2 are the same. So in this instance the code calling it would actually uninstall 3.0.0-beta1 and install 3.0.0-beta2 rather than upgrade.